be/src/runtime/runtime_state.cpp

```diff
@@ -556,4 +558,42 @@ bool RuntimeState::is_nereids() const {
     return _query_ctx->is_nereids();
 }
 
+std::vector<std::unique_ptr<RuntimeProfile>>& RuntimeState::pipeline_id_to_profile() {
```

Comment:
return value is a reference to the vector, after fucntion call, the vector will not be protected by the lock.